### PR TITLE
default transformer dropout to 0

### PR DESCRIPTION
**Pull request by JudeDavis1**
The reason for 0 dropout (at the pretraining/causal lm stage is that on limited hardware, you want the model to learn the dataset patterns quickly first, then use dropout at fine-tuning. Because dropout can slow down accuracy at the lm stage. So optional dropout may allow features to be quickly learned first.

### CODE REVIEW

**Comment by geohot** [comment]
Don't change behavior, but I'm okay with that being a parameter.

**Comment by JudeDavis1** [comment]
Ok, can change default to 0.1.

**Comment by JudeDavis1** [comment]
@geohot done
